test/sasinvariant/test/utest_data_handling.py

```diff
@@ -112,164 +111,195 @@ def test_initial_data_processing(self):
+#                    label='Original data', markersize=10)
+#          plt.loglog(inv._data.x[:npts],
+#                     inv._low_extrapolation_function.evaluate_model(inv._data.x[:npts]),
+#                     'r', label='Fitted line')
+#          plt.legend()
+#          plt.show()        
+
+#      self.assertEqual(qs_extr, _qstar+delta_qs_extr)
+#        self.assertAlmostEqual(dqs_extr, math.sqrt(dqstar*dqstar
+#                                                   + delta_dqs_extr
+#                                                   *delta_dqs_extr), 15)
+
         # We don't expect the extrapolated invariant to be very far from the
         # result without extrapolation. Let's test for a result within 10%.
-        self.assertTrue(math.fabs(qs_extr-qstar)/qstar<0.1)
-        
+#        self.assertTrue(math.fabs(qs_extr-qstar)/qstar<0.1)
+
+        # This section is just wrong in any case. PDB
         # Check that the two results are consistent within errors
         # Note that the error on the extrapolated value takes into account
-        # a systematic error for the fact that we may not know the shape of I(q) at low Q.
-        self.assertTrue(math.fabs(qs_extr-qstar)<dqs_extr)
-        
+        # a systematic error for the fact that we may not know the shape of
+        # I(q) at low Q.
+#        self.assertTrue(math.fabs(qs_extr-qstar)<dqs_extr)
+        pass
+
     def test_qstar_high_q(self):
         """
             Test high-q extrapolation
         """
         inv = invariant.InvariantCalculator(self.data)
-        
+
         # Basic sanity check
         _qstar = inv.get_qstar()
         qstar, dqstar = inv.get_qstar_with_error()
         self.assertEqual(qstar, _qstar)
-        
+
         # High-Q Extrapolation
         # Check that the returned invariant is what we expect given
         # the result we got without extrapolation
-        inv.set_extrapolation('high', npts=20, function='power_law')
+        inv.set_extrapolation('high', npts=75, function='power_law')
         qs_extr, dqs_extr = inv.get_qstar_with_error('high')
         delta_qs_extr, delta_dqs_extr = inv.get_qstar_high()
-        
-        # From previous analysis using SasView, we expect an exponent of about 3 
-        self.assertTrue(math.fabs(inv._high_extrapolation_function.power-3)<0.1)
-        
+
+        # In principle the slope should be -4. However due to the oscillations
+        # and finite number of points we need quite a lot of points to ensure
+        # that the fit averages close to 4. SasView estimates 3.92 at the
+        # moment 
+        self.assertTrue(math.fabs(inv._high_extrapolation_function.power-4)<0.1)
+
         self.assertEqual(qs_extr, _qstar+delta_qs_extr)
-        self.assertAlmostEqual(dqs_extr, math.sqrt(dqstar*dqstar + delta_dqs_extr*delta_dqs_extr), 10)
-        
+        self.assertAlmostEqual(dqs_extr, math.sqrt(dqstar*dqstar \
+                                        + delta_dqs_extr*delta_dqs_extr), 10)
```

Comment:
PEP8 discourages the use of `\` as a line extender (python gets confused sometimes if there is a space afterward, or if there is some problem with CRLF endings).  You do not need it within parentheses.

Better would be to break at a higher level semantic boundary:
```
self.assertAlmostEqual(
    dqs_extr, math.sqrt(dqstar*dqstar + delta_dqs_extr*delta_dqs_extr), 10)
```
or
```
self.assertAlmostEqual(
    dqs_extr, 
    math.sqrt(dqstar*dqstar + delta_dqs_extr*delta_dqs_extr), 
    10)
```

This appears in a number of places, not all of them noted in this review.